airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/NormalizationActivityImpl.java

```diff
@@ -62,13 +64,16 @@ public class NormalizationActivityImpl implements NormalizationActivity {
   private final WorkerEnvironment workerEnvironment;
   private final LogConfigs logConfigs;
   private final String airbyteVersion;
+  private final FeatureFlags featureFlags;
   private final Integer serverPort;
   private final AirbyteConfigValidator airbyteConfigValidator;
   private final TemporalUtils temporalUtils;
   private final ResourceRequirements normalizationResourceRequirements;
   private final AirbyteApiClient airbyteApiClient;
 
-  private final static Version MINIMAL_VERSION_FOR_DATATYPES_V1 = new Version("0.3.0");
+  private static final Version MINIMAL_VERSION_FOR_DATATYPES_V1 = new Version("0.3.0");
+  private static final String STRICT_COMPARISON_IMAGE_TAG = "0.4.0";
```

Comment:
Explicitly tagging versions here feels a bit fragile. This feels like a short term fix, do we have another solution in mind?

Reply:
Also, I think this value should be read from the config to be easier to override (ie not having to go through full build cycle to change the version here)

Reply:
can you say a bit more about the config? Not familiar with what you're referring to

Reply:
The value here being hardcoded means that to iterate on the tag we want to use when the feature flag is active would require merging another change in OSS.
Having it as env variables allow to patch the config we use to deploy and avoid the slowest part of the dev cycle.

Generally, configs for the worker should go in https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/resources/application.yml. The `${CONTAINER_ORCHESTRATOR_ENABLED:false}` notation is `${env_var:default}`.
This being a feature flag and the current constraints, it's probably fine to directly read the value from the featureflag environment.

Reply:
added flags for the new version (STRICT_COMPARISON_NORMALIZATION_TAG) and the version from which it was branched (NON_STRICT_COMPARISON_NORMALIZATION_TAG).

what do you think is better behavior if someone pushes a newer version of normalization (e.g. 0.2.26) without also updating the strict comparison version?
* use 0.2.26 (silently deactivate feature flag)
* use strict_comparison (silently use an older version of normalization)
* fail the sync
* other?

Reply:
My expectations in pseudo code would be:
```
if (shouldUseStrictComparisonForWorkspace(workspaceId)) {
  normalizationTag = FeatureFlag.getNormalizationTagForStrictComparision();
}
// the else clause should be implicitly keep using the version we defined in the connector catalog
// Same, the versioning logic should be applied on the tag without specific checks.
```

I think I'll look into cleaning up the datatypes v1 switch for the time being, so as long at we use data types v0 for anything that isn't in the `0.3.x` branch, we should be good.

Reply:
cool. pushed https://github.com/airbytehq/airbyte/pull/22381/commits/e470fd133a5902b3bdf8d2afc60736359860d60b, which I think accomplishes that.